ReactSkia/sdk/OnScreenKeyBoard.h

```diff
@@ -0,0 +1,175 @@
+ /*
+ * Copyright (C) 1994-2022 OpenTV, Inc. and Nagravision S.A.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+#ifndef OSK_H
+#define OSK_H
+
+#include <semaphore.h>
+
+#include "include/core/SkCanvas.h"
+#include "include/core/SkColor.h"
+
+#include "rns_shell/compositor/Compositor.h"
+#include "rns_shell/common/Window.h"
+#include "rns_shell/platform/graphics/PlatformDisplay.h"
+#include "rns_shell/platform/graphics/WindowContextFactory.h"
+
+#include "NotificationCenter.h"
+#include "RNSKeyCodeMapping.h"
+
+//Suported KeyBoards
+enum OSKTypes {
+    OSK_DEFAULT_TYPE,
+    OSK_ALPHA_NUMERIC_KB=OSK_DEFAULT_TYPE,
+    OSK_NUMERIC_KB,
+    OSK_TYPE_END
+};
+//Suported Themes
+enum OSKThemes {
+    OSK_DEFAULT_THEME,
+    OSK_DARK_THEME=OSK_DEFAULT_THEME,
+    OSK_LIGHT_THEME
+};
+// OSK Error Code
+enum OSKErrorCode {
+    OSK_ERROR_NONE,
+    OSK_ERROR_ANOTHER_INSTANCE_ACTIVE,
+    OSK_ERROR_LAUNCH_FAILED
+};
+enum KBLayoutType {
+    ALPHA_LOWERCASE_LAYOUT,
+    ALPHA_UPPERCASE_LAYOUT,
+    SYMBOL_LAYOUT,
+    NUMERIC_LAYOUT
+};
+
+enum keyType {
+    TOGGLE_KEY ,
+    TEXT_KEY,
+    FUNCTION_KEY,
+    KEY_TYPE_COUNT
+};
+
+//Configuration to be specified while OSK Launch
+struct OSKConfig {
+    OSKTypes        oskType;
+    OSKThemes       oskTheme;
```

Comment:
do we need to name the members as oskType or oskTheme , since they are already part of OSKConfig . can they be type , theme

Reply:
Will Change it